### PR TITLE
New integration

### PR DESCRIPTION
**Pull request by Rutu2k**
API Integration:
-Drives
-Problem Statement
-Rules
-CandidateForm
-CodeSubmission
-CustomIO

New Features: 
-Finish Button
-End page showing score and test time of candidate(dummy API)

### CODE REVIEW

**Review by atharva-borekar — APPROVED** [review]

File: src/actions/finishTestActions.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { FINISH_TEST } from 'constants/actionConstants';
+
+export const finishTestRequest = (data) => ({
+  type: FINISH_TEST.FINISH_TEST_REQUEST,
+  payload: { ...data },
+});
+
+export const finishTestSuccessAction = (responsedata) => ({
```
Comment (shailendrakanherkar18) [inline src/actions/finishTestActions.js:8]: responsedata => responseData

File: src/apis/problemStatementApi.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
 import request from 'apis/apiHelper';
 
-export const getStatement = (driveId) => request.get(`/problems/${driveId}`);
+export const getStatement = (driveId) => request.get('/drives/1/problem');
```
Comment (shailendrakanherkar18) [inline src/apis/problemStatementApi.js:3]: 1 ?? 
why we hardcoded ???

File: src/components/EditorNavComponent/index.jsx [hunk 3]
```diff
@@ -28,14 +31,37 @@ function EditorNavComponent({
...
+    if (isLoading) {
+      return (
+        <div className='overview-block d-flex text-center justify-content-center text-success '>
+          <Spinner size='sm' />
+        </div>
+      );
+    }
+  };
+
+  // const limitExceed = () => {
```
Comment (shailendrakanherkar18) [inline src/components/EditorNavComponent/index.jsx:51]: // const limitExceed = () => {
  //   if (limit) {
  //     console.log('hello');
  //     return <p>Limit Exceeded</p>;
  //   }
  // };

why commented ????
if we don't need it? please remove it.

File: src/components/EditorNavComponent/index.jsx [hunk 4]
```diff
@@ -58,21 +96,47 @@ function EditorNavComponent({
           ))}
         </DropdownMenu>
       </ButtonDropdown>
-      <Button className='custom-btn bg-color border-0' onClick={handleSubmit}>Submit</Button>
+      <div>
+        <Button className='custom-btn bg-color border-0' onClick={onSubmitClick}>
+          {limit ? (
```
Comment (shailendrakanherkar18) [inline src/components/EditorNavComponent/index.jsx:101]: {limit ? 'Limit Exceeded' : 'Submit' }

File: src/apis/problemStatementApi.js (hunk 2 above)
Comment (Rutu2k) [reply src/apis/problemStatementApi.js:3]: I forgot to fix it,(we were testing API). I will fix it

File: src/components/ProblemsComponent/index.jsx [hunk 5]
```diff
@@ -1,11 +1,20 @@
...
+import { Container, Spinner } from 'core-components';
 
 import './problemStyle.css';
 
-const ProblemComponent = ({ isError, errorMessage, title, description }) => {
+const ProblemComponent = ({ isError, errorMessage, title, description, isLoading }) => {
+  const loading = () => {
+    if (isLoading) {
+      return (
+        <div className='overview-block d-flex text-center justify-content-center text-success '>
```
Comment (shailendrakanherkar18) [inline src/components/ProblemsComponent/index.jsx:12]: Can we make this as a component ??

Comment (shailendrakanherkar18) [reply src/components/ProblemsComponent/index.jsx:12]: So it can be used at  couple of places.

File: src/sagas/finishtestSaga.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import { finishTestApi } from 'apis/finishTestApi';
+import { FINISH_TEST } from 'constants/actionConstants';
+import {
+  finishTestSuccessAction,
+  finishTestRequestFailed,
+} from 'actions/finishTestActions';
+
+export function* finishTestWorkerSaga(action) {
+  const { id, candidateId } = action.payload;
+  const obj = {
```
Comment (shailendrakanherkar18) [inline src/sagas/finishtestSaga.js:11]: Can we name it in better way like saying finishtesRequestPayload ???

File: src/components/ProblemsComponent/index.jsx (hunk 5 above)
Comment (Rutu2k) [reply src/components/ProblemsComponent/index.jsx:12]: yes!

**Review by shailendrakanherkar18 — APPROVED** [review]

**Pull request merged**